### PR TITLE
Added support for llvm.x86.mmx.punpckh[bw|wd|dq] and llvm.x86.mmx.punpckl[bw|wd|dq]

### PR DESCRIPTION
**Pull request by Stefan-Mada**
Added support for llvm.x86.mmx.punpckh[bw|wd|dq] and llvm.x86.mmx.punpckl[bw|wd|dq], as well as a set of tests to verify their operation.

### CODE REVIEW

File: ir/instr.cpp [hunk 1]
```diff
@@ -4023,6 +4041,35 @@ StateValue X86IntrinBinOp::toSMT(State &s) const {
...
+    case mmx_punpcklwd: laneCount = 4; startVal = 0; endVal = laneCount / 2; break;
+    case mmx_punpckldq: laneCount = 2; startVal = 0; endVal = laneCount / 2; break;
+    default: UNREACHABLE();
+    }
+    //Starts at first lane of high half of both vectors
+    for (unsigned i = startVal; i != endVal; ++i) {
+      auto ai = aty->extract(av, i);
+      auto bi = bty->extract(bv, i);
+      
+      vals.emplace_back(move(ai.value), move(ai.non_poison));
```
Comment (zhengyang92) [inline ir/instr.cpp]: just use `vals.emplace_back(move(ai));` to eliminate the redundant unpacking/packing.

File: ir/instr.h [hunk 2]
```diff
@@ -945,8 +945,8 @@ class ShuffleVector final : public Instr {
   std::unique_ptr<Instr> dup(const std::string &suffix) const override;
 };
 
-
```
Comment (zhengyang92) [inline ir/instr.h:948]: don't remove this line :)

File: ir/instr.cpp [hunk 3]
```diff
@@ -4023,6 +4041,35 @@ StateValue X86IntrinBinOp::toSMT(State &s) const {
...
+  case mmx_punpckhwd:
+  case mmx_punpckhdq:
+  case mmx_punpcklbw:
+  case mmx_punpcklwd:
+  case mmx_punpckldq:
+  {
+    vector<StateValue> vals;
+    unsigned laneCount, startVal, endVal;
+    switch (op) {
+    case mmx_punpckhbw: laneCount = 8; startVal = laneCount / 2; endVal = laneCount; break;
```
Comment (zhengyang92) [inline ir/instr.cpp:4054]: these lines are great! nice and clean.

**Review by zhengyang92 — COMMENTED** [review]
thanks :)

File: ir/instr.cpp (hunk 1 above)
Comment (Stefan-Mada) [reply ir/instr.cpp]: Beautiful. That makes it a lot cleaner.

**Pull request merged**